packages/esm-service-queues-app/src/queue-table/queue-table.component.tsx

```diff
@@ -0,0 +1,216 @@
+  const currentUser = useSession();
+  const { t } = useTranslation();
+
+  const currentLocationUuid = currentUser?.sessionLocation?.uuid;
+  const { visitQueueEntries: visitQueueEntriesByLocation, isLoading } =
+    useUnmappedVisitQueueEntries(currentLocationUuid);
+  const currentQueueUuid = useSelectedServiceUuid();
+  const { queues } = useQueues();
+  const allQueueAllowedStatuses = useCallback(() => {
+    const allStatusFromAllQueues = queues.flatMap((q) => q.allowedStatuses);
+    const statusByUuid = new Map<string, AllowedStatus>();
+    for (const status of allStatusFromAllQueues) {
+      statusByUuid.set(status.uuid, status);
+    }
+    return [...statusByUuid.values()];
+  }, [queues]);
+
+  const currentQueue = queues.find((q) => q.uuid === currentQueueUuid);
+  const allowedStatuses = currentQueue?.allowedStatuses ?? allQueueAllowedStatuses();
+
+  const [currentStatusUuid, setCurrentStatusUuid] = useState<string>(allowedStatuses?.[0]?.uuid);
+
+  useEffect(() => {
+    if (currentQueue) {
+      const newQueueHasCurrentStatus = currentQueue?.allowedStatuses?.some((s) => s?.uuid == currentStatusUuid);
+      if (!newQueueHasCurrentStatus) {
+        setCurrentStatusUuid(currentQueue.allowedStatuses?.[0]?.uuid);
+      }
+    }
+  }, [currentQueue]);
+
+  const currentStatusIndex = Math.max(0, allowedStatuses?.findIndex((s) => s?.uuid == currentStatusUuid));
+
+  const visitedQueueEntriesByLocationAndQueueAndStatus = visitQueueEntriesByLocation
+    ?.filter((entry) => !currentQueueUuid || entry.queueEntry.queue.uuid == currentQueueUuid)
+    ?.filter((entry) => !currentStatusUuid || entry.queueEntry.status.uuid == currentStatusUuid);
+
+  if (isLoading || !allowedStatuses?.length) {
+    return <>Loading....</>;
+  }
+
+  const handleServiceChange = ({ selectedItem }: { selectedItem: QueueServiceInfo }) => {
+    updateSelectedServiceUuid(selectedItem.uuid);
+  };
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.headerContainer}>
+        <div className={!isDesktop(layout) ? styles.tabletHeading : styles.desktopHeading}>
+          <h4>{t('patientsCurrentlyInQueue', 'Patients currently in queue')}</h4>
+        </div>
+      </div>
+      <div className={styles.filterContainer}>
+        <Dropdown
+          id="serviceFilter"
+          titleText={t('showPatientsWaitingFor', 'Show patients waiting for') + ':'}
+          label={currentQueue?.name ?? t('all', 'All')}
+          type="inline"
+          items={[{ display: `${t('all', 'All')}` }, ...queues]}
+          itemToString={(item) => (item ? item.display : '')}
+          onChange={handleServiceChange}
+          size="sm"
+        />
+      </div>
+      <Tabs
+        selectedIndex={currentStatusIndex}
+        onChange={({ selectedIndex }) => setCurrentStatusUuid(allowedStatuses[selectedIndex].uuid)}
+        className={styles.tabs}>
+        <TabList style={{ paddingLeft: '1rem' }} aria-label="Outpatient tabs" contained>
```

Comment:
It would be better to define a scss file and put this in a class (this should also be a standard Carbon spacing unit).